x-pack/plugins/security_solution/public/assistant/get_comments/stream/use_stream.test.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { renderHook } from '@testing-library/react-hooks';
+import { useStream } from './use_stream';
+
+const amendMessage = jest.fn();
+const reader = jest.fn();
+const cancel = jest.fn();
+const exampleChunk = {
+  id: '1',
+  object: 'chunk',
+  created: 1635633600000,
+  model: 'model-1',
+  choices: [
+    {
+      index: 0,
+      delta: { role: 'role-1', content: 'content-1' },
+      finish_reason: null,
+    },
+  ],
+};
+const readerComplete = {
+  read: reader
+    .mockResolvedValueOnce({
+      done: false,
+      value: new Uint8Array(new TextEncoder().encode(`data: ${JSON.stringify(exampleChunk)}`)),
+    })
+    .mockResolvedValueOnce({
+      done: false,
+      value: new Uint8Array(new TextEncoder().encode(``)),
+    })
+    .mockResolvedValueOnce({
+      done: false,
+      value: new Uint8Array(new TextEncoder().encode('data: [DONE]\n')),
+    })
+    .mockResolvedValue({
+      done: true,
+    }),
+  cancel,
+  releaseLock: jest.fn(),
+  closed: jest.fn().mockResolvedValue(true),
+} as unknown as ReadableStreamDefaultReader<Uint8Array>;
+
+const defaultProps = { amendMessage, reader: readerComplete };
+describe('useStream', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Should stream response. isLoading/isStreaming are true while streaming, isLoading/isStreaming are false when streaming completes', async () => {
```

Comment:
The description and implementation of this test appear to contain three assertions. Consider (generally) one assertion per test